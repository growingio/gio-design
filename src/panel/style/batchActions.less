@import (reference) '../../stylesheet/variables/index.less';
@panel-batch-actions-prefix-cls: ~'@{component-prefix}-panel__batch-actions';

.@{panel-batch-actions-prefix-cls} {
  display: flex;
  align-items: center;
  color: @gray-5;
  font-weight: normal;
  font-size: 14px;
  font-style: normal;
  &__text {
    margin-right: 4px;
  }

  &__close {
    margin-bottom: -1px;
  }

  &__divider {
    display: inline-block;
    width: 1px;
    height: 16px;
    margin-right: 4px;
    margin-left: 4px;
    background-color: @gray-2;
    border: 0;
  }

  &__children {
    & > * {
      margin-left: 12px;
    }
  }
}
