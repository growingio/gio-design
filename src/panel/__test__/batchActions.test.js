import React from 'react';
import { render } from '@testing-library/react';
import { Default as BatchActions } from '../batchActions.stories';

describe.skip('Render BatchActions with default args', () => {
  it('should render actions', () => {
    const wrapper = render(<BatchActions />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render batchActions', () => {
    const onClose = jest.fn();
    const wrapper = render(<BatchActions count={1} onClose={onClose} />);
    expect(wrapper).toMatchSnapshot();
  });
});
