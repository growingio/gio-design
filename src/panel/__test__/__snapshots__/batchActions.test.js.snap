// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BatchActions with default args should render actions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="gio-panel__batch-actions"
      >
        <span
          class="gio-panel__batch-actions__text"
        >
          已经选择 0 项
        </span>
        <button
          class="gio-btn gio-btn-text gio-btn-small gio-btn-icon-only"
          type="button"
        >
          <span
            aria-label="close-outlined"
            class="gio-icon"
            role="img"
          >
            <svg
              fill="currentColor"
              height="16px"
              viewBox="0 0 14 14"
              width="16px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <style />
              </defs>
              <g>
                <g>
                  <path
                    d="M6.62 5.92a.54.54 0 00.76 0l3.87-3.87a.49.49 0 01.7.7L8.08 6.62a.54.54 0 000 .76L12 11.25a.49.49 0 01-.7.7L7.38 8.08a.54.54 0 00-.76 0L2.75 12a.49.49 0 01-.7-.7l3.87-3.92a.54.54 0 000-.76L2.05 2.75a.49.49 0 01.7-.7z"
                    fill="currentColor"
                    fill-rule="evenodd"
                    id="close-outlined_svg__close"
                  />
                </g>
              </g>
            </svg>
          </span>
        </button>
        <span
          class="gio-panel__batch-actions__divider"
        />
        <span
          class="gio-panel__batch-actions__children"
        >
          <button
            class="gio-btn gio-btn-secondary"
            type="button"
          >
            <span>
              批量移动
            </span>
          </button>
          <button
            class="gio-btn gio-btn-secondary"
            type="button"
          >
            <span>
              批量删除 
            </span>
          </button>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="gio-panel__batch-actions"
    >
      <span
        class="gio-panel__batch-actions__text"
      >
        已经选择 0 项
      </span>
      <button
        class="gio-btn gio-btn-text gio-btn-small gio-btn-icon-only"
        type="button"
      >
        <span
          aria-label="close-outlined"
          class="gio-icon"
          role="img"
        >
          <svg
            fill="currentColor"
            height="16px"
            viewBox="0 0 14 14"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <style />
            </defs>
            <g>
              <g>
                <path
                  d="M6.62 5.92a.54.54 0 00.76 0l3.87-3.87a.49.49 0 01.7.7L8.08 6.62a.54.54 0 000 .76L12 11.25a.49.49 0 01-.7.7L7.38 8.08a.54.54 0 00-.76 0L2.75 12a.49.49 0 01-.7-.7l3.87-3.92a.54.54 0 000-.76L2.05 2.75a.49.49 0 01.7-.7z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  id="close-outlined_svg__close"
                />
              </g>
            </g>
          </svg>
        </span>
      </button>
      <span
        class="gio-panel__batch-actions__divider"
      />
      <span
        class="gio-panel__batch-actions__children"
      >
        <button
          class="gio-btn gio-btn-secondary"
          type="button"
        >
          <span>
            批量移动
          </span>
        </button>
        <button
          class="gio-btn gio-btn-secondary"
          type="button"
        >
          <span>
            批量删除 
          </span>
        </button>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Render BatchActions with default args should render batchActions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="gio-panel__batch-actions"
      >
        <span
          class="gio-panel__batch-actions__text"
        >
          已经选择 1 项
        </span>
        <button
          class="gio-btn gio-btn-text gio-btn-small gio-btn-icon-only"
          type="button"
        >
          <span
            aria-label="close-outlined"
            class="gio-icon"
            role="img"
          >
            <svg
              fill="currentColor"
              height="16px"
              viewBox="0 0 14 14"
              width="16px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <defs>
                <style />
              </defs>
              <g>
                <g>
                  <path
                    d="M6.62 5.92a.54.54 0 00.76 0l3.87-3.87a.49.49 0 01.7.7L8.08 6.62a.54.54 0 000 .76L12 11.25a.49.49 0 01-.7.7L7.38 8.08a.54.54 0 00-.76 0L2.75 12a.49.49 0 01-.7-.7l3.87-3.92a.54.54 0 000-.76L2.05 2.75a.49.49 0 01.7-.7z"
                    fill="currentColor"
                    fill-rule="evenodd"
                    id="close-outlined_svg__close"
                  />
                </g>
              </g>
            </svg>
          </span>
        </button>
        <span
          class="gio-panel__batch-actions__divider"
        />
        <span
          class="gio-panel__batch-actions__children"
        >
          <button
            class="gio-btn gio-btn-secondary"
            type="button"
          >
            <span>
              批量移动
            </span>
          </button>
          <button
            class="gio-btn gio-btn-secondary"
            type="button"
          >
            <span>
              批量删除 
            </span>
          </button>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="gio-panel__batch-actions"
    >
      <span
        class="gio-panel__batch-actions__text"
      >
        已经选择 1 项
      </span>
      <button
        class="gio-btn gio-btn-text gio-btn-small gio-btn-icon-only"
        type="button"
      >
        <span
          aria-label="close-outlined"
          class="gio-icon"
          role="img"
        >
          <svg
            fill="currentColor"
            height="16px"
            viewBox="0 0 14 14"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <style />
            </defs>
            <g>
              <g>
                <path
                  d="M6.62 5.92a.54.54 0 00.76 0l3.87-3.87a.49.49 0 01.7.7L8.08 6.62a.54.54 0 000 .76L12 11.25a.49.49 0 01-.7.7L7.38 8.08a.54.54 0 00-.76 0L2.75 12a.49.49 0 01-.7-.7l3.87-3.92a.54.54 0 000-.76L2.05 2.75a.49.49 0 01.7-.7z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  id="close-outlined_svg__close"
                />
              </g>
            </g>
          </svg>
        </span>
      </button>
      <span
        class="gio-panel__batch-actions__divider"
      />
      <span
        class="gio-panel__batch-actions__children"
      >
        <button
          class="gio-btn gio-btn-secondary"
          type="button"
        >
          <span>
            批量移动
          </span>
        </button>
        <button
          class="gio-btn gio-btn-secondary"
          type="button"
        >
          <span>
            批量删除 
          </span>
        </button>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
